mongoOperator/helpers/RestoreHelper.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2018 Ultimaker
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+import json
+import logging
+import os
+from base64 import b64decode
+from subprocess import check_output, CalledProcessError, SubprocessError
+
+from time import sleep
+from google.cloud.storage import Client as StorageClient
+from google.oauth2.service_account import Credentials as ServiceCredentials
+
+from mongoOperator.helpers.MongoResources import MongoResources
+from mongoOperator.models.V1MongoClusterConfiguration import V1MongoClusterConfiguration
+from mongoOperator.services.KubernetesService import KubernetesService
+# from mongoOperator.services.MongoService import MongoService
+
+
+class RestoreHelper:
+    """
+    Class responsible for handling the Restores for the Mongo cluster.
+    """
+    DEFAULT_BACKUP_PREFIX = "backups"
+    BACKUP_FILE_FORMAT = "mongodb-backup-{namespace}-{name}-{date}.archive.gz"
+    RESTORE_RETRIES = 4
+    RESTORE_WAIT = 15.0
+
+    def __init__(self, kubernetes_service: KubernetesService):
+        """
+        :param kubernetes_service: The kubernetes service.
+        """
+        self.kubernetes_service = kubernetes_service
+
+    def _getCredentials(self, cluster_object: V1MongoClusterConfiguration) -> dict:
+        """
+        Retrieves the storage credentials for the given cluster object from the Kubernetes secret as specified in the
+        cluster object.
+        :param cluster_object: The cluster object from the YAML file.
+        :return: The credentials dictionary.
+        """
+        secret_key = cluster_object.spec.backups.gcs.service_account.secret_key_ref
+        secret = self.kubernetes_service.getSecret(secret_key.name, cluster_object.metadata.namespace)
+        credentials_encoded = secret.data[secret_key.key]
+        credentials_json = b64decode(credentials_encoded)
+        return json.loads(credentials_json)
+
+    def getLastBackup(self, cluster_object: V1MongoClusterConfiguration) -> str:
```

Comment:
Maybe rename to getLastBackupStorageObjectName, as it doesn't actually return the backup itself?